### PR TITLE
fix(ci): Fix issue #8165 Aggregate PR comment on check failure 

### LINKED ISSUE #8164: Make comment gha workflow on check failure update last comment
When commenting on a PR automatically, a new comment get created each time, it would be best to have a single comment getting updated.

### LINKED ISSUE #8165: Aggregate PR comment on check failure
Find a way to not have one comment per check failure. But instead have a comment that get updated with the list of failure

### PR DESCRIPTION
**Pull request by ranj8work**
## Summary
Fix issue [8165](https://github.com/magma/magma/issues/8165) Aggregate PR comment on check failure 

## Test Plan
The testing is done on the forked repo master branch

## Additional Information

- [NO] This change is backwards-breaking

### CODE REVIEW

File: .github/workflows/comment-pr-on-check-failure.yml [hunk 1]
```diff
@@ -71,9 +75,46 @@ jobs:
...
+                                        repo: context.repo.repo,
+                                        issue_number: issue_number,
+                                      });
+
+            for (const c of commentsList['data']) {
+              oldMsg = c.body;
+              if( oldMsg.includes('Oops! Looks like you failed the ') && oldMsg.includes(process.env.WORKFLOW_NAME) ) {
+                  commentId = c.id;
+                  if( process.env.WORKFLOW_STATUS == 'failure' )  {
+                      updMsg = ":recycle: Updated: :x: The check is still failing the " + process.env.WORKFLOW_NAME + " after the last commit id";
```
Comment (quentinDERORY) [inline .github/workflows/comment-pr-on-check-failure.yml]: don't specify id if you don't give it

File: .github/workflows/comment-pr-on-check-failure.yml [hunk 2]
```diff
@@ -71,9 +75,46 @@ jobs:
...
+                  }
+                  else if( process.env.WORKFLOW_STATUS == 'success' ) {
+                      updMsg = ":recycle: Updated: :white_check_mark: The check is passing the " + process.env.WORKFLOW_NAME + " after the last commit id";
```
Comment (quentinDERORY) [inline .github/workflows/comment-pr-on-check-failure.yml]: here too

**Comment by quentinDERORY** [comment]
You cannot test this workflow on this PR, so you can remove the junk python file.

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
See comment

File: .github/workflows/comment-pr-on-check-failure.yml (hunk 2 above)
Comment (ranj8work) [reply .github/workflows/comment-pr-on-check-failure.yml]: @quentinDERORY resolved

**Comment by ranj8work** [comment]
> You cannot test this workflow on this PR, so you can remove the junk python file.

@quentinDERORY removed

File: .github/workflows/comment-pr-on-check-failure.yml [hunk 3]
```diff
@@ -37,7 +39,9 @@ jobs:
             });
             var fs = require('fs');
             fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(download.data));
-      - run: unzip pr.zip
+      - name: Unzip the pr zip file
+        if: ${{ github.event.workflow_run.conclusion == 'failure' }}
```
Comment (quentinDERORY) [inline .github/workflows/comment-pr-on-check-failure.yml]: remove the if, we always want to have the pr number at hand

**Review by quentinDERORY — CHANGES_REQUESTED** [review]
Always export the PR number

File: .github/workflows/comment-pr-on-check-failure.yml (hunk 3 above)
Comment (ranj8work) [reply .github/workflows/comment-pr-on-check-failure.yml:43]: @quentinDERORY - the pr.zip file is getting created only on failures from the other check failure workflows

Comment (ranj8work) [reply .github/workflows/comment-pr-on-check-failure.yml]: > remove the if, we always want to have the pr number at hand

@quentinDERORY  - Addressed this comment to propogate PR number on both failures and success. Pushed the changes.

**Review by quentinDERORY — APPROVED** [review]

**Pull request merged**